plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -171,3 +174,9 @@ class JvmAbiClassBuilderInterceptor : ClassGeneratorExtension {
             get() = internalName.endsWith(WhenByEnumsMapping.MAPPINGS_CLASS_NAME_POSTFIX)
     }
 }
+
+private fun shouldBeRemovedFromAbi(irClass: IrClass?, removePrivateTopLevelClasses: Boolean) = when {
+    irClass == null -> false
+    DescriptorVisibilities.isPrivate(irClass.visibility) -> removePrivateTopLevelClasses || !irClass.isTopLevel
```

Comment:
Should be

```suggestion
    DescriptorVisibilities.isPrivate(irClass.visibility) -> (irClass.isTopLevel && removePrivateTopLevelClasses) || !irClass.isTopLevel
```

Reply:
I need a second opinion on that. For me the shorter version is easier to read.

Reply:
I'm also for the shorter version